src/Sarif.Driver/Sdk/AnalyzeOptionsBase.cs

```diff
@@ -94,8 +94,18 @@ public abstract class AnalyzeOptionsBase
         public bool RichReturnCode { get; set; }
 
         [Option(
-            "persist-file-contents",
-            HelpText = "Persist a base64-encoded representation of all referenced files to the log.")]
-        public bool PersistFileContents { get; set; }
+            't', // persist 't'ext
+            "persist-text-contents",
+            HelpText = "Persist a base64-encoded representation of all referenced textual files to the log.")]
+        public bool PersistTextFileContents { get; set; }
+
+        [Option(
+            // Lack of shortcut for this one is by design, as the need
+            // to persist this data is unusual and could lead to insecurities.
```

Comment:
Is it insecure because it's easier for a bad actor to modify the binary data without anyone noticing?

Reply:
no. generally, it's just a 'Bad Idea' to carry binary data around (due to the possibility someone will execute it).

---
In reply to: [200718546](https://github.com/Microsoft/sarif-sdk/pull/936#discussion_r200718546) [](ancestors = 200718546)